sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuTransitionOverrides.scala

```diff
@@ -407,31 +407,50 @@ class GpuTransitionOverrides extends Rule[SparkPlan] {
     }
   }
 
+  // find
+  // GpuDataWritingCommandExec(GpuHashJoin)
+  // GpuDataWritingCommandExec(GpuHashAggregateExec)
+  // GpuDataWritingCommandExec(GpuProject(GpuHashJoin))
+  // GpuDataWritingCommandExec(GpuProject(GpuHashAggregateExec))
```

Comment:
I don't understand why this is looking for very specific patterns leading to a write.  It seems likely to break because we forgot some pattern or some new pattern shows up later.  Note the original comment says:
```
// Insert a sort after the last hash-based op before the query result if there are no
// intermediate nodes that have a specified sort order.
```
meaning we don't care about any nodes between the hashing node and the writing node if they don't change the sort order.  The original code did not require that the hashing node appeared immediately before the write, but this effectively does (with some hardcoded exceptions for a project).

Reply:
Yes, you are right, updated the code to follow the original logic.
Please review again.